src/Generator.cpp

```diff
@@ -996,10 +998,19 @@ void execute_generator(const ExecuteGeneratorArgs &args_in) {
     // -------------- Do some sanity checking.
     internal_assert(!args.output_dir.empty());
 
+    const bool hlpipe_only = args.output_types.size() == 1 &&
+                             args.output_types.count(OutputFileType::hlpipe) == 1;
+
     const bool cpp_stub_only = args.output_types.size() == 1 &&
                                args.output_types.count(OutputFileType::cpp_stub) == 1;
-    if (!cpp_stub_only) {
-        // It's ok to leave targets unspecified if we are generating *only* a cpp_stub
+
+    const bool hlpipe_and_cpp_stub_only = args.output_types.size() == 2 &&
```

Comment:
Is there a real-world use case for emitting these two and nothing else? I can't think of one. Unless we know of one, I don't know why we should special case it. (Especially since hlpipe requires a Target, unlike cppstub, so they don't really go together)

Reply:
Yes, I've separated these, and now handle the hlpipe separately within a target specific context.